### PR TITLE
DataSource: implement Send

### PR DESCRIPTION
**Pull request by jlgale**
We're trying to do connection pooling in our application and we discovered that we can't `Send` `DataSource` handles.

Raw pointers don't implement `Send` by default (`Raii<ffi::Dbc>` is the culprit) so I just added it. (See https://doc.rust-lang.org/nomicon/send-and-sync.html for background) I'm not very familiar with the details of the ODBC library so I don't know of reason why this could be a problem, but it seems to work for us.

I'm not a very strong rust programmer, so you may know of a better way.

### CODE REVIEW

**Comment by Koka** [comment]
@jlgale I'm not 100% sure it's safe thing to do. 
Have you tried to just wrap DataSource in something like `Arc<Mutex<DataSource>>` to safely pass it between threads. This will ensure that no thread access datasource simultaneously and that everything will be dropped properly at the end. @pacman82 what do you think?

**Comment by pacman82** [comment]
That is a tough one. Wether this is safe or not depends on three things: The driver manager (e.g. unix-odbc), the driver and the configured threading level. There is now way to find out at compile time if all the requirements are met. I have seen people spending days debugging only to find out that they need to configure threading level = 2 for this particular driver.
TL;DR it is not guaranteed to be safe.

Yet, often it is perfectly fine to send `DataSource`s to other threads. I would suggest the following: The default `DataSource` type remains unsendable. Yet by calling an unsafe method it can be promoted to a type which implements send. The documentation of these method could contain some hints for the user which configurations to check. I think having the user explicitly executing one unsafe operation is the right thing to do, since the compiler can not check the safety guarantees itself and the user needs to be aware that it is his responsibility to do so.

**Comment by jlgale** [comment]
@Koka @pacman82 thanks, we resolved this issue by implementing `Send` on our own type, as you suggested. So we don't need this PR anymore. I'll close it.
